tests/end_to_end/end_to_end.cpp

```diff
@@ -169,37 +235,61 @@ typedef class _udp_packet
         udp->destPort = destination_port;
     }
 
-  private:
-    void
-    set_mac_addresses(_In_ const std::array<uint8_t, 6>& source_mac, _In_ const std::array<uint8_t, 6>& destination_mac)
+} udp_packet_t;
+
+typedef class _ip_in_ip_packet : public ip_packet_t
+{
+  public:
+    _ip_in_ip_packet(
+        ADDRESS_FAMILY address_family,
+        _In_ const std::array<uint8_t, 6>& source_mac = _test_source_mac,
+        _In_ const std::array<uint8_t, 6>& destination_mac = _test_destination_mac,
+        _In_opt_ const void* outer_ip_addrs = nullptr,
```

Comment:
avoid abbreviations ("addrs")
```suggestion
        _In_opt_ const void* outer_ip_addresses = nullptr,
```